src/kvstore/kvstore_local.h

```diff
@@ -61,7 +61,10 @@ class KVStoreLocal : public KVStore {
   }
 
   virtual ~KVStoreLocal() {
-    delete comm_;
```

Comment:
I think `delete nullptr` is safe

Reply:
Ok, I will change that to only add `comm_ = nullptr;` after deleting. I added the check because the previous version of the destructor was not setting comm_ to nullptr and it gave me segfault when kvstore_nccl called both its own destructor and kvstore_local destructor (both trying to delete comm_).